sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,876 @@
+to ``sol``.
+
+Step 5 : Find the poles and their multiplicities of `a(x)`. Let
+the number of poles be `n`. Also find the valuation of `a(x)` at
+`\infty` using ``val_at_inf``.
+
+NOTE: Although the algorithm considers `\infty` as a pole, it is
+not mentioned if it a part of the set of finite poles. `\infty`
+is NOT a part of the set of finite poles. If a pole exists at
+`\infty`, we use its multiplicty to find the laurent series of
+`a(x)` about `\infty`.
+
+Step 6 : Find `n` c-vectors (one for each pole) and 1 d-vector using
+``construct_c`` and ``construct_d``. Now, determine all the ``2**(n + 1)``
+combinations of choosing between 2 choices for each of the `n` c-vectors
+and 1 d-vector.
+
+NOTE: The equation for `d_{-1}` in Case 4 (Pg 80) has a printinig
+mistake. The term `- d_N` must be replaced with `-N d_N`. The same
+has been explained in the code as well.
+
+For each of these above combinations, do
+
+Step 8 : Compute `m` in ``compute_m_ybar``. `m` is the degree bound of
+the polynomial solution we must find for the auxiliary equation.
+
+Step 9 : In ``compute_m_ybar``, compute ybar as well where ``ybar`` is
+one part of y(x) -
+
+.. math:: \overline{y}(x) = \sum_{i=1}^{n} \sum_{j=1}^{r_i} \frac{c_{ij}}{(x - x_i)^j} + \sum_{i=0}^{N} d_i x^i
+
+Step 10 : If `m` is a non-negative integer -
+
+Step 11: Find a polynomial solution of degree `m` for the auxiliary equation.
+
+There are 2 cases possible -
+
+    a. `m` is a non-negative integer: We can solve for the coefficients
+    in `p(x)` using Undetermined Coefficients.
+
+    b. `m` is not a non-negative integer: In this case, we cannot find
+    a polynomial solution to the auxiliary equation, and hence, we ignore
+    this value of `m`.
+
+Step 12 : For each `p(x)` that exists, append `ybar + \frac{p'(x)}{p(x)}`
+to ``sol``.
+
+Step 13 : For each solution in ``sol``, apply an inverse transformation,
+so that the solutions of the original equation are found using the
+solutions of the equation in its normal form.
+"""
+
+
+from itertools import product
+from sympy.core import S
+from sympy.core.add import Add
+from sympy.core.numbers import oo
+from sympy.core.function import count_ops
+from sympy.core.relational import Eq
+from sympy.core.symbol import symbols, Symbol, Dummy
+from sympy.functions import sqrt, exp
+from sympy.functions.elementary.complexes import sign
+from sympy.integrals.integrals import Integral
+from sympy.polys.domains import ZZ
+from sympy.polys.polytools import Poly
+from sympy.polys.polyroots import roots
+from sympy.solvers.solveset import linsolve
+
+
+def riccati_normal(w, x, b1, b2):
+    """
+    Transforming the solution of a Riccati ODE
+
+    w'(x) = b0 + b1*w(x) + b2*w(x)^2 ... (1)
+
+    to the solution of its corresponding normal
+    Riccati ODE
+
+    y'(x) + y(x)^2 = a(x)            ... (2)
+
+    If w(x) is the solution to (1), then the
+    solution to (2) is
+
+    y(x) = -b2(x)*w(x) - b2'(x)/(2*b2(x)) - b1(x)/2
+    """
```

Comment:
This docstring doesn't say what the function actually returns or specify what the parameters are.

It could be something like:
```
Reduce a Riccati ODE given by b1 and b2 to normal form given by a.

A general Riccati ODE has the form

.. math:: w'(x) = ...

The substitution $w(x) = ...$ leads to a normal form Riccati ODE

.. math: y'(x) + y(x)^2 = a(x)

Given b1 and b2 this function returns a for the normal form ODE. A solution `y1`
can be transformed to a solution of the original ODE with

.. math: y1(x) = w1(x)...
```

Reply:
Actually, this function does not return the normal form of the Riccati ODE. It applies the transformation on the solution for the Riccati equation which gives the solution to the corresponding normal Riccati ODE.

Reply:
Oh I see. Well the docstring should still explain what the parameters and return value of the function are.